crates/cervo-runtime/src/runtime.rs

```diff
@@ -85,6 +105,44 @@ impl Runtime {
 
     /// Executes all models with queued data.
     pub fn run(&mut self) -> Result<HashMap<BrainId, HashMap<AgentId, Response<'_>>>, CervoError> {
+        #[cfg(feature = "threaded")]
+        {
+            Ok(self.run_threaded())
+        }
```

Comment:
We should offer users option to do whatever they want here. So `run` + `run_for` are always available and single-threaded. `run_threaded` and `run_threaded_for` are guarded by `#[cfg(feature = "threaded"]`. No magic branching internally - easier APIs.